lib/src/intl_suggestors/intl_importer.dart

```diff
@@ -121,12 +141,23 @@ _InsertionLocation _insertionLocationForPackageImport(
     // No directive to insert relative to; insert before the first member or
     // at the beginning of the file.
     return _InsertionLocation(unit.declarations.firstOrNull?.offset ?? 0,
-        trailingNewlineCount: 2);
+        trailingNewlineCount: 2, usePackageImports: hasPackageImports);
+  }
+
+  for (final importDirective in imports) {
```

Comment:
This could be more concisely written as something like

bool hasRelativeImports(imports) imports.any((importDirective) => importDirective.uri != null && importDirective.uri.scheme != 'package' && importDirective.uri.scheme != 'dart');

Reply:
Code updated like 
```
 for (final importDirective in imports) {
    final uriContent = importDirective.uri.stringValue;
    if (uriContent != null) {
      final uri = Uri.parse(uriContent);
      hasPackageImports = !imports.any((importDirective) =>
          uri != null && uri.scheme != 'package' && uri.scheme != 'dart:');
    }
  }
```

Reply:
Now you're looping over imports twice. If you have the added line for hasPackageImports =, then you don't need the outer for loop.

Reply:
code updated